src/gui/qgsmaptoolcapture.cpp

```diff
@@ -289,10 +291,40 @@ bool QgsMapToolCapture::tracingAddVertex( const QgsPointXY &point )
     mSnappingMatches.append( QgsPointLocator::Match() );
   }
 
+  // If the layer supports curves, we de-approximate curves
+  QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( mCanvas->currentLayer() );
+  if ( vlayer->dataProvider()->capabilities().testFlag( QgsVectorDataProvider::Capability::CircularGeometries ) )
+  {
+    // We convert the capture curve to OGR geometry
+    OGRGeometry *ogrCaptureCurve;
+    OGRGeometryFactory::createFromWkt( mCaptureCurve.asWkt().toLocal8Bit().data(), nullptr, &ogrCaptureCurve );
+
+    // We de-approximate the curves using OGR (the geometry must be forced to linear first, as a mCaptureCurve is a CompoundCurve)
+    OGRGeometry *ogrLinestring = ogrCaptureCurve->getLinearGeometry(); // it must be forced to linear, as captuecurve is a compound curve
```

Comment:
You can't use the gdal c++ api -- you need to use the c api equivalents.

Reply:
Ok (done) but just out of interest, why ? (to me the C++ api definitely looks easier to read)

Reply:
C API is stable (same for GEOS for example: provides long-term ABI stability)